api/src/main/java/org/pentaho/platform/api/genericfile/exception/OperationWithPathFailedException.java

```diff
@@ -0,0 +1,62 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.platform.api.genericfile.exception;
+
+import org.pentaho.platform.api.genericfile.GenericFilePath;
+
+/**
+ * The exception class thrown when an Operation has failed, and we need to have a path to give more context of the
+ * error.
+ */
+public class OperationWithPathFailedException extends OperationFailedException {
```

Comment:
I think this exception class should be abstract, given it is not expected to be directly thrown, afaict.